uefi/src/table/boot.rs

```diff
@@ -793,6 +793,19 @@ impl BootServices {
         (self.reinstall_protocol_interface)(handle, protocol, old_interface, new_interface).into()
     }
 
+    /// Installs one or more protocol interfaces to the boot services environment.
+    ///
+    /// # Errors
+    ///
+    /// See section `EFI_BOOT_SERVICES.InstallMultipleProtocolInterfaces()` in the UEFI Specification for more details.
+    ///
+    /// * [`uefi::Status::ALREADY_STARTED`]
+    /// * [`uefi::Status::INVALID_PARAMETER`]
+    /// * [`uefi::Status::OUT_OF_RESOURCES`]
+    pub fn install_multiple_protocol_interfaces(&self, handle: Handle, mut args: ...) -> Result {
```

Comment:
I think we should either mark this as unsafe, as we do not guarantee any type-safety here, or do not support the UEFI function at all but build a custom solution around `install_protocol_interface`, which is well typed.

What do you think, @nicholasbishop ?

Reply:
Is it possible to construct a [VaList](https://doc.rust-lang.org/nightly/core/ffi/struct.VaList.html) in Rust? I don't see a way after looking at the docs a bit but I'm not that familiar with the feature. Basically what I'm thinking is, we'd have the safe wrapper here take a slice of `(&Guid, *mut c_void)` pairs, then convert that into a valist for calling the underlying UEFI function.

If we can't do that, then I think marking this function `unsafe` is the right thing to do.

Incidentally, I have previously wondered why `InstallMultipleProtocolInterfaces` matters since it seems like you could just call `InstallProtocolInterface` multiple times. https://uefi.org/specs/UEFI/2.10/07_Services_Boot_Services.html#driver-model-boot-services says:
> The third group of boot services is designed to help simplify the implementation of drivers, and produce drivers with smaller executable footprints. [...] The [EFI_BOOT_SERVICES.InstallMultipleProtocolInterfaces()](https://uefi.org/specs/UEFI/2.10/07_Services_Boot_Services.html#efi-boot-services-installmultipleprotocolinterfaces) and [EFI_BOOT_SERVICES.UninstallMultipleProtocolInterfaces()](https://uefi.org/specs/UEFI/2.10/07_Services_Boot_Services.html#efi-boot-services-uninstallmultipleprotocolinterfaces) are very useful to driver writers. These boot services allow one or more protocol interfaces to be added or removed from a handle. In addition, InstallMultipleProtocolInterfaces() guarantees that a duplicate device path is never added to the handle database. This is very useful to bus drivers that can create one child handle at a time, because it guarantees that the bus driver will not inadvertently create two instances of the same child handle.

Reply:
> I think we should either mark this as unsafe, as we do not guarantee any type-safety here, or do not support the UEFI function at all but build a custom solution around `install_protocol_interface`, which is well typed.
> 
> What do you think, @nicholasbishop ?

This can potentially a solution. But I will look into passing variadic arguments for this function.